src/components/select/filterable-select/filterable-select.component.js

```diff
@@ -281,13 +282,14 @@ const FilterableSelect = React.forwardRef(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [value, onChange, children]);
 
+    const onOpen = useStableCallback(onOpenProp);
```

Comment:
I was wondering why you hadn't used `useStableCallback` on `onOpen` in multi-select - and looking at the code I see why, it already handles this by using a callback argument to `setOpenState` to ensure we don't call `onOpen` if it's already open. I'm wondering if you considered using that approach in filterable-select too? It seems odd to me that these two components which are so similar are using completely different ways to control the calling of `onOpen`, and this may have been an opportunity to make them more consistent with each other. I think both approaches are fine, and the `useStableCallback` is probably simpler to write, but really I think we should use the same one in both components.

Reply:
Using the same pattern as in multi-select would require creating a new method which then should be added as a dependency to other methods, meaning I'd have to wrap it in `useStableCallback` either way to prevent some unnecessary renders or method executions. Its obvious that it is possible to implement it in a cleaner way (like extracting some of the methods to a separate `useSelect` hook to reduce redundancy) but I'd rather do it in a separate dedicated `Refactor Select components` task.